app/models/katello/content_view.rb

```diff
@@ -111,6 +111,27 @@ class ContentView < Katello::Model
     scoped_search :on => :composite, :complete_value => true
     scoped_search :on => :generated_for, :complete_value => true
     scoped_search :on => :default # just for ordering
+    scoped_search :on => :name, :complete_value => true,
+                  :rename => :content_views,
+                  :operators => ['=', '!=', '~', '!~'],
+                  :ext_method => :find_components_by_cv_name
+
+    def self.find_components_by_cv_name(key, operator, value)
+      kcv = Katello::ContentView.table_name
+      kcvc = Katello::ContentViewComponent.table_name
+      { :conditions => "#{kcv}.composite = 't' AND #{kcv}.id IN (SELECT #{kcvc}.composite_content_view_id FROM #{kcvc} WHERE #{kcvc}.content_view_id IN (SELECT #{kcv}.id FROM #{kcv} WHERE #{kcv}.name #{operator} ?))",
+        :parameter => [value],
+      }
+    end
+
+    def self.completer_scope_options(search)
+      if search.include?('content_views')
+        # Don't autocomplete CCV names when searching for components
+        { :value_filter => { :composite => false } }
+      else
+        {}
+      end
+    end
```

Comment:
Autocomplete was including composite CV names in the suggested search. For example, if I was searching for `content_views = xxx` and had typed `content_views = `, the suggestions would include all of my content view names, including _composite_ content views. But a composite content view cannot contain itself, so this made no sense. To prevent this I added this method, which modifies scoped_search's value filter. I had to modify it to pass in the search because otherwise it would filter out composite CVs when I searched for `name = `, which we don't want.